lib/intercom/request.rb

```diff
@@ -70,6 +70,13 @@ def execute(target_base_url=nil, username:, secret: nil)
             parsed_body = parse_body(decoded_body, response)
             raise_errors_on_failure(response)
             parsed_body
+          rescue Intercom::RateLimitExceeded => e
+            if @handle_rate_limit
```

Comment:
We'll need a max retry mechanism here. Unlikely scenario but we can risk running into an infinite loop if we don't have one.

Reply:
makes sense, any suggestion on max retry? 10 or 20 or something like that?

Reply:
Yup. We wait for a fair bit here so I would go with something small like 3. This operation risks getting too long otherwise. Thoughts @mmartinic ?

Reply:
3 sounds good. we can make this configurable later